RandBLAS/dense.hh

```diff
@@ -68,33 +68,29 @@ struct DenseDist {
     const int64_t n_cols;
 };
 
-template <typename T>
+template <typename T, typename RNG = r123::Philox4x32>
 struct DenseSkOp {
-    const DenseDist dist;
-    const RNGState seed_state;
-    RNGState next_state;
-    const bool own_memory = true;
-    /////////////////////////////////////////////////////////////////////
-    //
-    //      Properties specific to dense sketching operators
-    //
-    /////////////////////////////////////////////////////////////////////
 
-    T *buff = nullptr;
-    bool filled = false;
-    bool persistent = true;
-    const blas::Layout layout = blas::Layout::ColMajor;
+    using generator = RNG;
+    using state_type = RNGState<RNG>;
+    using buffer_type = T;
```

Comment:
Why do you need to define ``using buffer_type = T;``? My guess is so that we can template w.r.t ``typename SKOP`` and then in implementations that work with a sketching operator ``S`` we access the numerical type with ``using T = S::buffer_type;``. Is that correct?

Reply:
yes, it's a way of getting the type of data stored, from with in templated code.

Reply:
Amazing! God, I'm so glad you're part of this project. I knew this was an important thing that I wanted to do, but since I didn't know how to describe it, I didn't google it effectively.